### DISCUSSION TITLE
Joystick Adapter audio out

### DISCUSSION THREAD

**Discussion by IMSAI-Oracle** [thehighnibble/thehighnibble.github.io#148] [S-132 add-on for IMSAI & Cromemco replicas]
Question, how would I go about getting audio out of the joystick adapter card to a speaker?

![Joystick Adapter](https://github.com/thehighnibble/thehighnibble.github.io/assets/131218791/529886c0-6322-4f93-92ca-38b6f8afdfa8)

Thanks,
Lynn

**Comment by nbreeden2** [comment]
The Analog-GND to Digital-GND jumper should be installed. In your photo I see it is (upper left corner).

Due to limitations of the ESP-32 only 2 analog outputs are implemented on the S-132. A1 and A3. These match the two analog outputs designated "Audio" for the joysticks in the Cromemco Docs (A1 and A3).

My attachment shows the view down into the 2x20 connector orientated as shown in your attached photo. The 2x20 connector is broken up into 4 sections of 1x10 pins, each section is color coded in my attachment.

My attachment also shows the emulated ports used (in grey), these match the original defaults for the D+7AIO card from Cromemco.

The virtual D+7AIO needs to be enabled in the S-132 menus. There are three settings in the menu for how the D+7AIO is emulated. These settings have to do with the internal emulator configuration and help limit the amount of ESP-32 cycles the emulation will use.

Keep in mind that the original D+7AIO used bi-polar A2Ds / D2As. The original D2A output voltages could swing from approximately -2.4 to +2.4 volts. The original A2D input voltage range is also approximately -2.4 to +2.4 volts. These are represented digitally as an 8-bit 2's compliment value.

Because the ESP-32 is a 3.2V part and no negative voltages are used the voltage input/output range is approximately -0.0 to +3.2 volts. This means the emulation both shifts the voltage range up and provides a narrower range.

D+7AIO  -2.4 to +2.4V (The actual S-100 PCB)
Emulator 0.0 to +3.2V

The emulator also uses an 8-bit 2's compliment value. The emulation scales this in such a way that it remaps the original -2.4V to +2.4V range to a 0.0V to 3.2V range.

I demonstrate some programming against the joysticks and buttons in C (BDSC compiler) on my YouTube channel. There are 3 videos in the series, it starts with the one linked. Note that I misspeak in these videos. The 8 channels of A2D functionality on the S-132 is provided by an A2D external to the ESP-32.

https://youtu.be/skwqI-xDGBM

I will eventually post a video on programming the analog outputs. I have an actual D+7AIO card that I want to get working so I can demonstrate the output from an actual card to the emulated card. I don't know when I'll get this video shot and posted.

I hope this helps. -Neil

![ProgrammingModel](https://github.com/thehighnibble/thehighnibble.github.io/assets/37715892/d66f1474-6631-47ff-8890-25a53b2bac5d)

**Reply by IMSAI-Oracle** [reply]
Neil,


Thank you very much for the very detailed response. I have very much enjoyed your videos (I’ve been a subscriber for a while) anyhow your videos on the S-132 sent me down this path, to connect my original Cameco JS-1 to my IMSAI 8080 replica.


Using your information from the breakout of the 2×20 pin connector, I was able to trace the audio out to the following connectors.


AudioPatch connector pin out:

Pin 1 – DAC .A
Pin 2 -  Pin 1 of the Amp connector
Pin 3 – DAC .B


Amp connector pin out:

Pin 1 – Center pin of the AudioPatch
Pin 2 - + 5V
Pin 3 – ground

Lynn
